utils/releaser/src/ReleaseWorker/ReleaseCandidate/ValidateRequireFormatInComposerJsonReleaseWorker.php

```diff
@@ -0,0 +1,139 @@
+
+    /**
+     * @var bool
+     */
+    private $isSuccessful = false;
+
+    /**
+     * @param \Symplify\MonorepoBuilder\FileSystem\ComposerJsonProvider $composerJsonProvider
+     * @param \Symplify\MonorepoBuilder\FileSystem\JsonFileManager $jsonFileManager
+     */
+    public function __construct(
+        ComposerJsonProvider $composerJsonProvider,
+        JsonFileManager $jsonFileManager
+    ) {
+        $this->composerJsonProvider = $composerJsonProvider;
+        $this->jsonFileManager = $jsonFileManager;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     * @return string
+     */
+    public function getDescription(Version $version): string
+    {
+        return 'Validate "require" and "require-dev" version format for all packages';
+    }
+
+    /**
+     * Higher first
+     * @return int
+     */
+    public function getPriority(): int
+    {
+        return 920;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     */
+    public function work(Version $version): void
+    {
+        foreach ($this->composerJsonProvider->getRootAndPackageFileInfos() as $fileInfo) {
+            $jsonContent = $this->jsonFileManager->loadFromFileInfo($fileInfo);
+
+            $this->validateVersions($jsonContent, 'require', $fileInfo);
+            $this->validateVersions($jsonContent, 'require-dev', $fileInfo);
+        }
+
+        if ($this->isSuccessful) {
+            $this->symfonyStyle->success(Message::SUCCESS);
+        } else {
+            $this->symfonyStyle->confirm('Confirm there are the same package versions in each package');
+        }
+    }
+
+    /**
+     * @param mixed[] $jsonContent
+     * @param string $section
+     * @param \Symfony\Component\Finder\SplFileInfo $splFileInfo
+     */
+    private function validateVersions(array $jsonContent, string $section, SplFileInfo $splFileInfo): void
+    {
+        if (!isset($jsonContent[$section])) {
+            return;
+        }
+
+        foreach ($jsonContent[$section] as $packageName => $version) {
+            if ($this->shouldSkipPackageNameAndVersion($packageName, $version)) {
+                continue;
+            }
+
+            $this->symfonyStyle->warning(sprintf(
+                '"%s" file has invalid version format for "%s": "%s"',
+                $splFileInfo->getPathname(),
+                $packageName,
+                $version
+            ));
+
+            $this->isSuccessful = false;
+        }
+    }
+
+    /**
+     * @param string $packageName
+     * @param string $version
+     * @return bool
+     */
+    private function shouldSkipPackageNameAndVersion(string $packageName, string $version): bool
+    {
+        if (Strings::startsWith($packageName, 'ext-')) {
+            return true;
+        }
+
+        if (Strings::startsWith($version, '^')) {
+            return true;
+        }
+
+        // skip shopsys packages mutual dependencies
+        if (Strings::startsWith($packageName, 'shopsys')) {
```

Comment:
it should allow the packages from monorepo only, not all dependencies starting with "shopsys" (e.g. "shopsys/doctrine-orm": "2.6.2" is not reported at the moment but it should be)